File: qinglian-app-root/src/main/java/com/yychainsaw/pojo/vo/UserVO.java
package com.yychainsaw.pojo.vo;

import io.swagger.v3.oas.annotations.media.Schema;


import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Schema(description = "UserVO模型")
public class UserVO {
    @Schema(description = "userId")
    private UUID userId;
    private String username;
    @Schema(description = "nickname")
    private String nickname;
    private String avatarUrl;
    @Schema(description = "height")
    private Integer height;
    private BigDecimal weight;
    @Schema(description = "gender")
    private String gender;
    private LocalDateTime lastLoginTime;
    // 注意：绝对不包含 passwordHash
}

File: QingLian-app-android/app/src/main/java/com/yychainsaw/qinglianapp/data/model/vo/UserVO.kt
package com.yychainsaw.qinglianapp.data.model.vo

import com.google.gson.annotations.SerializedName

data class UserVO(
    @SerializedName("userId", alternate = ["user_id", "id"])
    val userId: String,

    val username: String,
    val nickname: String?,

    @SerializedName("avatarUrl", alternate = ["avatar", "headImg", "avatar_url"])
    val avatarUrl: String?,

    val gender: String?,

    // 修复：增加 "height_cm" (数据库原名) 和 "heightCm" 以防后端映射偏差
    @SerializedName("height", alternate = ["height_cm", "heightCm"])
    val height: Double?,

    // 修复：增加 "weight_kg" (数据库原名) 和 "weightKg"
    @SerializedName("weight", alternate = ["weight_kg", "weightKg"])
    val weight: Double?,

    // 增加 "last_login_time" 以匹配 SQL 原字段名
    @SerializedName("lastLoginTime", alternate = ["last_login_time"])
    val lastLoginTime: String?
)